lib/cadet_web/controllers/assets_controller.ex

```diff
@@ -0,0 +1,90 @@
+defmodule CadetWeb.AssetsController do
+  use CadetWeb, :controller
+
+  use PhoenixSwagger
+  alias Cadet.Stories.Assets
+
+  def index(conn, _params = %{"foldername" => folder_name}) do
+    with :ok <- Assets.validate_assets_role(conn),
+         :ok <- Assets.validate_folder_name(folder_name) do
+      assets = Assets.list_assets(folder_name)
+      render(conn, "index.json", assets: assets)
+    else
+      {:error, {status, message}} -> conn |> put_status(status) |> text(message)
+    end
+  end
+
+  @spec delete(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def delete(conn, _params = %{"folderName" => folder_name, "filename" => filename}) do
+    with :ok <- Assets.validate_assets_role(conn),
+         :ok <- Assets.validate_folder_name(folder_name) do
+      Assets.delete_object(folder_name, filename)
+      render(conn, "s3_response.json", resp: :ok)
+    else
+      {:error, {status, message}} -> conn |> put_status(status) |> text(message)
+    end
+  end
+
+  def upload(conn, %{"upload" => upload_params, "details" => details}) do
+    {:ok, details} = Jason.decode(details)
+    %{"folderName" => folder_name} = details
+
+    filename =
+      if Map.has_key?(details, "filename") do
+        details["filename"]
+      else
+        upload_params.filename
+      end
+
+    with :ok <- Assets.validate_assets_role(conn),
+         :ok <- Assets.validate_folder_name(folder_name),
+         :ok <- Assets.validate_filetype(filename) do
+      resp = Assets.upload_to_s3(upload_params, folder_name, filename)
+      render(conn, "s3_response.json", resp: resp)
```

Comment:
Could just return the string here. [`json`](https://hexdocs.pm/phoenix/Phoenix.Controller.html#json/2)